pkg/rules/engine/engine.go

```diff
@@ -63,45 +62,10 @@ func NewEngine(sigs []detect.Signature, sources EventSources, output chan detect
 	engine.signaturesIndex = make(map[detect.SignatureEventSelector][]detect.Signature)
 	engine.signaturesMutex.Unlock()
 	for _, sig := range sigs {
-		engine.signaturesMutex.Lock()
-		engine.signatures[sig] = make(chan protocol.Event, engine.config.SignatureBufferSize)
-		engine.signaturesMutex.Unlock()
-		meta, err := sig.GetMetadata()
+		_, err := engine.loadSignature(sig)
 		if err != nil {
-			engine.logger.Printf("error getting metadata: %v", err)
-			continue
+			engine.logger.Printf("%v", err)
 		}
-		se, err := sig.GetSelectedEvents()
-		if err != nil {
-			engine.logger.Printf("error getting selected events for signature %s: %v", meta.Name, err)
-			continue
-		}
-		for _, es := range se {
-			if es.Name == "" {
-				es.Name = ALL_EVENT_TYPES
-			}
-			if es.Origin == "" {
-				es.Origin = ALL_EVENT_ORIGINS
-			}
-			if es.Source == "" {
-				engine.logger.Printf("signature %s doesn't declare an input source", meta.Name)
-			} else {
-				engine.signaturesMutex.Lock()
-				engine.signaturesIndex[es] = append(engine.signaturesIndex[es], sig)
-				engine.signaturesMutex.Unlock()
-			}
-		}
-		err = sig.Init(engine.matchHandler)
-		if err != nil {
-			engine.logger.Printf("error initializing signature %s: %v", meta.Name, err)
-			continue
-		}
-	}
-	engine.signaturesMutex.RLock()
-	lenSigs := len(engine.signatures)
-	engine.signaturesMutex.RUnlock()
-	if len(sigs) != lenSigs {
-		return nil, errors.New("one or more signatures are not uniquely identifiable")
```

Comment:
Is this check removed intentionally? I think it's useful to catch issue when someone accidentally used the same identifier for two different signatures.

Reply:
I did remove this on purpose, mostly because it was screwing tests, but I also think it's more correct this way since duplicate signatures shouldn't stop construction IMO, and this case will be logged due to this error in line 218:
```
// signature already exists
return "", fmt.Errorf("failed to store signature: signature already exists")
```

WDYT? Should we bring this back anyway?

Reply:
Personally, I would fail fast rather than troubleshoot why one signature was overwritten accidentally. Such errors shows issues in input signatures that must be fixed anyway.

Reply:
There wouldn't be an overwrite, the signature already exists and so the load fails.

Reply:
Got it. Would be nice to add test scenario for such case one day.